ee/secureenclavesigner/secureenclavesigner_test.go

```diff
@@ -0,0 +1,171 @@
+//go:build darwin
+// +build darwin
+
+package secureenclavesigner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/kolide/kit/ulid"
+	"github.com/kolide/krypto/pkg/challenge"
+	"github.com/kolide/krypto/pkg/echelper"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	testWrappedEnvVarKey = "SECURE_ENCLAVE_TEST_WRAPPED"
+	macOsAppResourceDir  = "./test_app_resources"
+)
+
+func WithBinaryPath(p string) opt {
+	return func(ses *secureEnclaveSigner) {
+		ses.pathToLauncherBinary = p
+	}
+}
+
+// TestSecureEnclaveTestRunner TODO: description
+// #nosec G306 -- Need readable files
+func TestSecureEnclaveSigner(t *testing.T) {
+	t.Parallel()
+
+	if os.Getenv("CI") != "" {
+		t.Skipf("\nskipping because %s env var was not empty, this is being run in a CI environment without access to secure enclave", testWrappedEnvVarKey)
+	}
+
+	// set up app bundle
+	rootDir := t.TempDir()
+	appRoot := filepath.Join(rootDir, "launcher_test.app")
+
+	// make required dirs krypto_test.app/Contents/MacOS and add files
+	require.NoError(t, os.MkdirAll(filepath.Join(appRoot, "Contents", "MacOS"), 0777))
+	copyFile(t, filepath.Join(macOsAppResourceDir, "Info.plist"), filepath.Join(appRoot, "Contents", "Info.plist"))
+	copyFile(t, filepath.Join(macOsAppResourceDir, "embedded.provisionprofile"), filepath.Join(appRoot, "Contents", "embedded.provisionprofile"))
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	serverPrivKey, err := echelper.GenerateEcdsaKey()
+	require.NoError(t, err)
+
+	serverPubKeyDer, err := echelper.PublicEcdsaToB64Der(serverPrivKey.Public().(*ecdsa.PublicKey))
+	require.NoError(t, err)
+
+	// build the executable
+	executablePath := filepath.Join(appRoot, "Contents", "MacOS", "launcher_test")
+	out, err := exec.CommandContext( //nolint:forbidigo // Only used in test, don't want as standard allowcmd
```

Comment:
```suggestion
	out, err := exec.CommandContext( //nolint:forbidigo // Only used in test, don't want as standard allowedcmd
```

I think this comment is missing a word also -- same on 156.